protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/StepContainer.tsx

```diff
@@ -193,6 +201,17 @@ export function StepContainer(props: StepContainerProps): JSX.Element {
     cancel: cancelDelete,
   } = useConditionalConfirm(handleDelete, true)
 
+  const handleOpenForm = (clickNum: number, e: ReactMouseEvent): void => {
+    if (hasTrash) {
+      if (clickNum === 1) {
+        onClick?.(e)
+      } else {
+        onDoubleClick?.(e)
+      }
+    } else {
+      makeSnackbar(t('trash_required') as string)
+    }
+  }
```

Comment:
probably we can avoid using a nested if-statement by changing the check order.

```suggestion
    if (!hasTrash) {
      makeSnackbar(t('trash_required') as string)
    }
    
     if (clickNum === 1) {
        onClick?.(e)
      } else {
        onDoubleClick?.(e)
      }
  }
```